identity-core/src/common/timestamp.rs

```diff
@@ -165,6 +188,49 @@ impl Diff for Timestamp {
   }
 }
 
+/// A span of time.
+///
+/// This type is typically used to increment or decrement a [`Timestamp`].
+#[derive(Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord, Deserialize, Serialize)]
+#[repr(transparent)]
+pub struct Duration(time::Duration);
+
+// copy a small subset of the functionality offered by time::Duration. By not copying everything it will be easier
+// to migrate to another implementation if that will be necessary in the future.
+// Do not allow negative Durations hence all constructors take u32 instead of i64, this makes it possible for us to
+// switch to relying on (for instance) std::time::Duration if we want to.
```

Comment:
Nit: condense comment slightly.

```suggestion
// Re-expose a small subset of time::Duration and use u32 instead of i64 
// to disallow negative durations. This gives us the flexibility to migrate
// the internal representation to e.g. std::time::Duration in the future
// if required.
```